src/ui/public/vislib/visualizations/point_series/column_chart.js

```diff
@@ -155,25 +172,39 @@ export function VislibVisualizationsColumnChartProvider(Private) {
       const yScale = this.getValueAxis().getScale();
       const groupCount = this.getGroupedCount();
       const groupNum = this.getGroupedNum(this.chartData);
-      const groupSpacingPercentage = 0.15;
+      const gutterSpacingPercentage = 0.15;
       const isTimeScale = this.getCategoryAxis().axisConfig.isTimeDomain();
       const isHorizontal = this.getCategoryAxis().axisConfig.isHorizontal();
       const isLogScale = this.getValueAxis().axisConfig.isLogScale();
       const minWidth = 1;
       let barWidth;
+      let gutterWidth;
 
       if (isTimeScale) {
         const { min, interval } = this.handler.data.get('ordered');
-        let groupWidth = xScale(min + interval) - xScale(min);
-        groupWidth = Math.abs(groupWidth);
-        const groupSpacing = groupWidth * groupSpacingPercentage;
+        let intervalWidth = xScale(min + interval) - xScale(min);
+        intervalWidth = Math.abs(intervalWidth);
 
-        barWidth = (groupWidth - groupSpacing) / groupCount;
+        gutterWidth = intervalWidth * gutterSpacingPercentage;
+        barWidth = (intervalWidth - gutterWidth) / groupCount;
       }
 
-      function x(d) {
+      /**
+       * Histogram intervals are not always equal widths, e.g, monthly time intervals.
```

Comment:
yah, I was tried creating a common function. The problem is that if you are outside of the scope of `addStackedBars ` or `addGroupedBars ` then the number of arguments balloons and the function loses readability since it is mainly just a bunch of arguments.